### ISSUE TITLE
Compatibility with air / Positron

### ISSUE THREAD

**Issue by lorenzwalthert** [r-lib/styler#1292]
The Positron IDE seems to have removed support for {styler}, so if you want to use the IDEs formatting capabilities, you can't use {styler} (anymore). This raises the question (again) for compatibility between the two formatters. #1291 shows that once formatted with air, styler will hardly chance anything due to its non-invasiveness but initial diff is large. Hence, that means using format on save or other IDE features from positron is not conflicting in particular with the pre-commit hook for {styler}.

**Comment by lorenzwalthert** [comment]
If we wanted to use {styler} in Positron, we'd probably have to create a VS Code extension for this sole purpose. I think the existing R vs code extension conflicts with positron and can't be used.

**Comment by lorenzwalthert** [comment]
Sounds like a good project for AI assisted  / agentic coding: Narrow scope with good documentation, small scope, easy-ish harness. I won't work on it though 😀. Here's GitHub Copilot output:

Creating a VS Code extension that exposes an R package call (like `styler`) for formatting is moderately complex, but absolutely achievable. Let me break down the key considerations:

## Complexity Overview

**Difficulty Level: Medium** — it's not trivial but well-documented territory. The main challenges are:

1. **R environment detection** — Finding the user's R installation
2. **Process execution** — Running R scripts/code reliably
3. **Error handling** — Capturing and displaying R errors gracefully
4. **Cross-platform compatibility** — Windows/Mac/Linux differences

## Minimal Implementation Path

The simplest approach would be a **document formatter provider** that:

```javascript
// 1. Call R via bash: `R --slave -e "styler::style_file('path/to/file')"`
// 2. Read the formatted file
// 3. Apply changes to the editor
```

## What You'd Need

### Core Extension Files
- `extension.ts` — Main extension logic
- `package.json` — Configuration, contributes formatters
- A utility to spawn child processes and run R commands

### Key Decisions

| Aspect | Options |
|--------|---------|
| **Invocation** | On-save, command palette, format selection |
| **R Detection** | Prompt for R path, auto-detect common locations, use `which R` |
| **Error Display** | Show in VS Code diagnostics, output channel, notification |
| **Temp Files** | Work with originals or temp files? |

## Rough Effort Estimate

- **Basic version** (formats current file): 4-6 hours
- **Polished version** (settings, error handling, multi-platform): 1-2 days
- **Production-ready** (tests, edge cases, marketplace): 2-3 days

## Quick Start Resources

The VS Code extension API is excellent. Look at:
- [VS Code Language Server Extension Guide](https://code.visualstudio.com/api/language-extensions/overview)
- [Document Formatter Provider](https://code.visualstudio.com/api/language-extensions/programmatic-language-features#provide-formatting) (easiest path)
- Examples: [prettier-vscode](https://github.com/prettier/prettier-vscode), [r-lsp](https://github.com/REditorSupport/vscode-R)

Would you like me to help you scaffold a basic extension, or do you have specific questions about any of these aspects?

**Comment by lorenzwalthert** [comment]
I could not resist the temptation to see if Kiro could help me build something like this in a few hours, and it seems yes. Experimental, and live at https://open-vsx.org/extension/lorenzwalthert/styler.
